firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/cobs.txt

```diff
@@ -0,0 +1,131 @@
+Scenario: The cobs::decode method correctly decodes encoded buffers
+
+ GIVEN("The Util COBS::decode function")
+
+  WHEN("The COBS::decode method is called on a buffer containing '0x01' as the only byte")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is empty")    
+
+  WHEN("The COBS::decode method is called on a buffer containing '0x03' as the only byte")      
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+     
+  WHEN("The COBS::decode method is called on a buffer that contains these bytes '0x05 0x02 0x03' ")    
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+
+  WHEN("The COBS::decode method is called on a buffer that contains these bytes '0x05 0x02 0xff' ")    
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+
```

Comment:
We need a test that captures the essence of the test we had been troubleshooting last call - since these three above WHENs the first byte is too big compared to the length of the input buffer, I don't think these WHENs cover the case we were looking at last time.

Reply:
`0x02 0x02 0xff` has been added as a simplified one, let's also add `0xff 0x10 ... 0x10 0x02 0x10 0x10` as the exact example which we were getting confused by.